shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -381,6 +384,19 @@ class WebFrameRenderer : public gin::Wrappable<WebFrameRenderer>,
 
   void OnDestruct() override {}
 
+  // Avoid using DidCreateScriptContext since it crashes if an exception is
```

Comment:
What does it crash on? I'm not sure how i feel here about hooking `DidInstallConditionalFeatures` here since it leaves a hole for potential races and other issues given that it's called in several callsites where `DidCreateScriptContext` isn't called, e.g. https://github.com/electron/electron/blob/97b353a30a0e92a169fb0557ddff641fa0332d7c/shell/renderer/electron_render_frame_observer.cc#L75

Reply:
I looked into this again and it actually works fine now just to use `DidCreateScriptContext`.

The error I received before was related to a CHECK which failed when the microtask queue was empty. I believe the microtask scope added here had fixed it:
https://github.com/electron/electron/blob/e87ef3c82d92ac8599c9c0a13f50ac34240f046c/shell/renderer/api/electron_api_web_frame.cc#L387-L391